model/lifecycle.go

```diff
@@ -1442,14 +1442,26 @@ func AddNewTasks(ctx context.Context, activated bool, v *Version, p *Project, pa
 		// a record in the TVPairSet indicating that it should exist
 		tasksToAdd := []string{}
 		for _, taskname := range pairs.ExecTasks.TaskNames(b.BuildVariant) {
-			if _, ok := existingTasksIndex[taskname]; ok {
+			if t, ok := existingTasksIndex[taskname]; ok {
+				if !t.Activated && activated {
+					// update task activation
+					if _, err = t.ActivateTask(evergreen.User); err != nil {
```

Comment:
should/do we also address the case where a new task depends on an existing task in another build?
Also, if we're creating a [new build](https://github.com/evergreen-ci/evergreen/blob/916f88e7fa43911b48ba15d115cb3826c3be9c08/model/generate.go#L261) with tasks that depend on other builds, will their dependencies be activated?

Reply:
I believe both of those cases are handled by IncludeDependencies, as we add dependencies before we split into new/existing variants.